Source/ARTPush.m

```diff
@@ -224,12 +224,18 @@ + (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error rest:(
 }
 
 - (void)activate {
+#if TARGET_OS_IOS && !TARGET_OS_SIMULATOR
+    [[UIApplication sharedApplication] registerForRemoteNotifications];
+#endif
     [self getActivationMachine:^(ARTPushActivationStateMachine *stateMachine) {
         [stateMachine sendEvent:[ARTPushActivationEventCalledActivate new]];
     }];
 }
 
 - (void)deactivate {
+#if TARGET_OS_IOS && !TARGET_OS_SIMULATOR
+    [[UIApplication sharedApplication] unregisterForRemoteNotifications];
```

Comment:
Consider enforcing that this is only called from main thread - see https://github.com/ably/ably-cocoa/pull/1293#issuecomment-1061804417.

Reply:
https://github.com/ably/ably-cocoa/pull/1293/commits/b8093726b8075edb39f83341b18745cc0ed29d43